datalab/datalab_session/file_utils.py

```diff
@@ -6,44 +6,41 @@
 from fits2image.conversions import fits_to_jpg, fits_to_tif
 
 from datalab.datalab_session.exceptions import ClientAlertException
-from datalab.datalab_session.s3_utils import get_fits, add_file_to_bucket
+from datalab.datalab_session.s3_utils import save_fits_and_thumbnails
 
 log = logging.getLogger()
 log.setLevel(logging.INFO)
 
-def get_hdu(basename: str, extension: str = 'SCI', source: str = 'archive') -> list[fits.HDUList]:
+def get_hdu(path: str, extension: str = 'SCI') -> list[fits.HDUList]:
```

Comment:
Similarly, I think a lot of these get_hdu, get_fits, etc... could live in a class that abstracts them away into a very simple, testable interface. I'm having trouble following what a few of these functions do and how they should be called, so maybe we can meet about this tomorrow?